components/calendar/src/chinese_based.rs

```diff
@@ -52,16 +51,22 @@ pub(crate) trait ChineseBased: CalendarArithmetic + Sized {
     /// reflect traditional methods of year-tracking or eras, since Chinese-based calendars
     /// may not track years ordinally in the same way many western calendars do.
     const EPOCH: RataDie;
-
-    /// Given a year, month, and day, create a Date<C> where C is a Chinese-based calendar.
-    ///
-    /// This function should just call try_new_C_date where C is the name of the calendar.
-    fn new_chinese_based_date(year: i32, month: u8, day: u8) -> ChineseBasedDateInner<Self>;
 }
 
 /// Chinese-based calendars define DateInner as a calendar-specific struct wrapping ChineseBasedDateInner.
-#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
-pub(crate) struct ChineseBasedDateInner<C: ChineseBased>(pub(crate) ArithmeticDate<C>);
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub(crate) struct ChineseBasedDateInner<C: ChineseBased>(
+    pub(crate) ArithmeticDate<C>,
+    pub(crate) ChineseBasedCache,
+);
+
+/// A caching struct used to store information for ChineseBasedDates
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub(crate) struct ChineseBasedCache {
+    pub(crate) new_year: RataDie,
+    pub(crate) is_leap_year: bool,
+    pub(crate) leap_month: u8,
```

Comment:
Suggestion (optional): you can merge these fields into one with `Option<NonZeroU8>`